src/app/components/shared/audio-event-card/annotation-event-card.component.ts

```diff
@@ -4,35 +4,41 @@ import {
   CUSTOM_ELEMENTS_SCHEMA,
   ElementRef,
   input,
-  OnInit,
-  ViewChild,
+  viewChild,
 } from "@angular/core";
 import { MediaControlsComponent } from "@ecoacoustics/web-components/@types/components/media-controls/media-controls";
 import { Annotation } from "@models/data/Annotation";
 import { FaIconComponent } from "@fortawesome/angular-fontawesome";
+import { SpectrogramComponent } from "@ecoacoustics/web-components/@types";
+import { LoadingComponent } from "@shared/loading/loading.component";
 import { ZonedDateTimeComponent } from "../datetime-formats/datetime/zoned-datetime/zoned-datetime.component";
 import { IsUnresolvedPipe } from "../../../pipes/is-unresolved/is-unresolved.pipe";
 
 @Component({
   selector: "baw-annotation-event-card",
-  templateUrl: "annotation-event-card.component.html",
-  styleUrl: "annotation-event-card.component.scss",
-  imports: [FaIconComponent, ZonedDateTimeComponent, IsUnresolvedPipe],
+  templateUrl: "./annotation-event-card.component.html",
+  styleUrl: "./annotation-event-card.component.scss",
+  imports: [
+    FaIconComponent,
+    ZonedDateTimeComponent,
+    IsUnresolvedPipe,
+    LoadingComponent,
+  ],
   schemas: [CUSTOM_ELEMENTS_SCHEMA],
 })
-export class AnnotationEventCardComponent implements OnInit, AfterViewInit {
+export class AnnotationEventCardComponent implements AfterViewInit {
   public annotation = input.required<Annotation>();
 
-  protected spectrogramId: string;
+  // Note that there is no { static: true } option for viewChild signals.
+  // This is on purpose, because signals are supposed to be able to
+  // automatically detect non-changes.
+  // https://github.com/angular/angular/issues/54376
+  private mediaControls =
+    viewChild<ElementRef<MediaControlsComponent>>("mediaControls");
+  private spectrogram =
+    viewChild<ElementRef<SpectrogramComponent>>("spectrogram");
 
-  @ViewChild("mediaControls", { static: true })
-  private mediaControls: ElementRef<MediaControlsComponent>;
-
-  public ngOnInit(): void {
-    this.spectrogramId = `spectrogram-${this.annotation().id}`;
-  }
-
-  public ngAfterViewInit(): void {
-    this.mediaControls.nativeElement.for = this.spectrogramId;
+  public ngAfterViewInit() {
+    this.mediaControls().nativeElement.for = this.spectrogram().nativeElement;
```

Comment:
i don't suppose we could do this step in the template:

`<oe-media-controls [for]="this.spectrogram.nativeElement">`

Reply:
I wish we could do this (this was my original hope for allowing element reference's in `for`).

However, element reference's in the template are `undefined` until after the first render (`AfterViewInit`).

And when the element reference finally becomes defined, it doesn't trigger Lit's rendering cycle.

Reply:
> it doesn't trigger Lit's rendering cycle.

really? that seems like something to unpack

Again, not going to insist on this, but it seems like something we should understand